backend/app/routers/authentication.py

```diff
@@ -206,6 +206,42 @@ async def revoke_admin(
         )
 
 
+@router.post("/users/enable_readonly/{useremail}", response_model=UserOut)
+async def enable_readonly_user(
+    useremail: str, current_username=Depends(get_current_user), admin=Depends(get_admin)
+):
+    if admin and current_username.admin:
```

Comment:
Should this be `if admin and current_username.admin_mode:`? Otherwise you are checking the same thing twice.